### ISSUE TITLE
[BUG] tf.while_loop creates a seg-fault when setting parallel_iterations to high values

### ISSUE THREAD

**Issue by georgh** [tensorflow/tensorflow#14623]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: both
- **TensorFlow version (use command below)**: 1.3 / 1.4
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: 0.5.4
- **GCC/Compiler version (if compiling from source)**: 5.4.0
- **CUDA/cuDNN version**: 8
- **GPU model and memory**: Tesla P100-PCIE-16GB
- **Exact command to reproduce**:
### Describe the problem

The following code crashes on all of my test systems, with tensorflow 1.3 or 1.4.
Doesn't matter if build with MKL or without or using the pip-version.
```
import numpy as np
import tensorflow as tf

num_dim = 20
FORMAT = tf.float64 #32 or 64 does not matter
n = 3500 #reducing the number results in normal execution

def simpeLoop(alpha):
      i = tf.constant(0)
      m0 = tf.zeros([num_dim, 1], dtype=FORMAT)
      cond = lambda i, m: i < n
      def body(ic, vec): 
            #a meaningless example, summing up the first num_dim elements of a vector 
            op = alpha[ic * num_dim:(ic + 1) * num_dim, :]
            # with tf.control_dependencies([op]): #if you uncomment this, it will not fault!
            return ic + 1, vec + op
      loop = tf.while_loop(cond, body, [i, m0], parallel_iterations=10**4, back_prop=False)
      return loop[1]

with tf.device('/cpu:0'):
      alpha = tf.placeholder(FORMAT, [None, 1], name="alpha")
      fdict = {
          alpha: np.random.rand(n * num_dim, 1),
      }
      op = simpeLoop(alpha)
      op = tf.reduce_sum(op) #not necessary for the seg fault
      init = tf.global_variables_initializer()

config = tf.ConfigProto()
threads = 1
config.intra_op_parallelism_threads = threads
config.inter_op_parallelism_threads = threads
sess = tf.Session(config=config)
sess.run(init, feed_dict=fdict)
print("init")
print(sess.run(op, feed_dict=fdict))
```
Adding the control_dependencies results in normal execution.

The documentation states 

> The maximum number of parallel iterations can be controlled by parallel_iterations, which gives users some control over memory consumption and execution order

so I would expect parallel_iterations to be some kind of upper bound. 
Small tests like presented in  #12937 showed that increasing the parallel_iterations number results in higher performance. But setting it to high results in a seg fault.

**Comment by carlthome** [comment]
Makes sense. The execution engine probably runs into a stack overflow or something. `10**4` is a lot!

Just for fun, you could plot the runtime in seconds as a function of `parallell_iterations` from 1 to the max before you segfault? 😃

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by georgh** [comment]
I guess its still an issue but I didn't have time to make the plot.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by georgh** [comment]
Digging a bit into the core code, It might be that the implementation of executor.cc is the problem:
The [FrameState](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/executor.cc#L935
) has an attribute called iterations, which is always resized to max_parallel_iterations:
[define](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/executor.cc#L1001) and [resize](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/executor.cc#L2310).
Iterations is a [gtl::inlined_vector](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/lib/gtl/inlined_vector.h#L330) which sets its internal storage when resized. 

Iterations is only a vector of pointer, so 10**4 should not be a problem. 
But maybe thats a starting point of someone with better overview of the internal of tf tries to fix this bug.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tatatodd** [comment]
Assigning to @skye, who knows much more about control flow than I do.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 91 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 106 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 121 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 136 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 151 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 166 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 181 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 196 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jvishnuvardhan** [comment]
This is a stale issue. Please check the issue with latest TensorFlow. If the issue still persists in the newer version of TF, please feel free to reopen it by providing details about the issue and a standalone code to reproduce the issue. Thanks!
